### DISCUSSION TITLE
Can not connect to remote via ssh on Mac

### DISCUSSION THREAD

**Discussion by celticwebdesign** [spatie/ray#588] [Q&A]
I am able to SSH connect to a remote server via the Mac Terminal command line, but when attempting to setup and connect Ray to remote server 'WPEngine' (WordPress managed hosting), I receive error notifications.

Port 21:
"Ray timed out while waiting for a handshake from your server. Are your credentials correct and can you SSH to your server?"

Port 22:
"Ray was unable to bind to the remote server's port. Is the port in use or is someone else using ray on this server?"

I have spoken with WPEngine about this but they cannot help further as the normal SSH connection via Terminal is working.

I would be grateful for any thoughts on how to make this connection via the Ray app.

**Comment by AdrianMrn** [comment]
@AlexVanderbist @freekmurze any idea?

**Reply by AlexVanderbist** [reply]
@celticwebdesign can you ask your contact at WPEngine if they have any limitations in place for SSH Port Forwarding or SSH Tunneling?

**Reply by celticwebdesign** [reply]
Thanks, I will ask them right now, might be 24hrs before I get anything back.

**Reply by celticwebdesign** [reply]
@AlexVanderbist I have received further information from WPEngine:

"That is a great question! So SSH port forwarding is not supported on our SSH gateway. Since our SSH setup uses a sidecar to connect to the server, there are some limitations.

This section of our SSH Gateway article goes over other limitations of SSH Gateway: https://wpengine.com/support/ssh-gateway/#SSH_Gateway_Limitations"

Does that help, would you like me to ask additional questions?
Thanks

**Reply by AlexVanderbist** [reply]
@celticwebdesign that's a shame. Ray uses SSH port forwarding to forward the Ray port on the server to your local Ray instance. Without port forwarding, I'm afraid remote servers wont work...

**Reply by celticwebdesign** [reply]
@AlexVanderbist 
Ok thanks for your help.
I will see if there is anything further that WPEngine can do.
